mailbox/api/src/test/java/org/apache/james/mailbox/events/EventBusContract.java

```diff
@@ -97,17 +128,48 @@ default void dispatchShouldNotThrowWhenAGroupListenerFails() {
             .doesNotThrowAnyException();
     }
 
+    @Test
+    default void dispatchShouldKeepWorkingIfGroupListenerFails() {
+        EventCallsVerifierMailboxListener listener = newEventCallsListener();
+        eventBus().register(listener, new GroupA());
+
+        doThrow(new RuntimeException())
+            .doCallRealMethod()
+            .when(listener).event(any());
+
+        eventBus().dispatch(EVENT, NO_KEYS).block();
+        eventBus().dispatch(EVENT, NO_KEYS).block();
+
+        WAIT_CONDITION
+            .until(() -> assertThat(listener.numberOfEventCalls()).isEqualTo(1));
+    }
+
     @Test
     default void dispatchShouldNotThrowWhenARegisteredListenerFails() {
         MailboxListener listener = newListener();
         doThrow(new RuntimeException()).when(listener).event(any());
 
         eventBus().register(listener, KEY_1);
 
-        assertThatCode(() -> eventBus().dispatch(EVENT, NO_KEYS).block())
+        assertThatCode(() -> eventBus().dispatch(EVENT, KEY_1).block())
             .doesNotThrowAnyException();
     }
 
+    @Test
+    default void dispatchShouldKeepWorkingARegisteredListenerFails() {
+        EventCallsVerifierMailboxListener listener = newEventCallsListener();
+        doThrow(new RuntimeException())
+            .doCallRealMethod()
+            .when(listener).event(any());
+        eventBus().register(listener, KEY_1);
+
+        eventBus().dispatch(EVENT, KEY_1).block();
+        eventBus().dispatch(EVENT, KEY_1).block();
+
+        WAIT_CONDITION
+            .until(() -> assertThat(listener.numberOfEventCalls()).isEqualTo(1));
+    }
+
```

Comment:
Well... If you are launched on the topic of exception handling, we need to check that a failing listener do not prevent other listeners from being executed.

Proposal for group:

```
eBus.register(failingListener, groupA);
eBus.register(listener, groupB);

eBus.dispatch(event, No_KEYS).block();

verify(listener, ...).event(any()); // listener is still executed after failingListener failure
```